src/test/java/com/answerdigital/benhession/academy/answerkingweek2/services/OrderServiceTest.java

```diff
@@ -0,0 +1,376 @@
+
+    @Test
+    void testAddOrderReturnsSavedOrder() {
+        // Given
+        Order order = new Order("42 Main St");
+
+        // When
+        when(orderRepository.save(any()))
+                .thenReturn(order);
+
+        // Then
+        assertSame(order, orderService.addOrder("42 Main St"));
+        verify(orderRepository).save(any());
+    }
+
+    @Test
+    void testFindByIdReturnsFoundOrder() {
+        // Given
+        Order order = new Order("42 Main St");
+
+        // When
+        when(orderRepository.findById(anyLong()))
+                .thenReturn(Optional.of(order));
+
+        // Then
+        assertSame(order, orderService.findById(123L));
+        verify(orderRepository).findById(anyLong());
+    }
+
+    @Test
+    void testFindByIdThrowsNotFoundException() {
+        // When
+        when(orderRepository.findById(anyLong())).thenReturn(Optional.empty());
+
+        // Then
+        assertThrows(NotFoundException.class, () -> orderService.findById(123L));
+        verify(orderRepository).findById(anyLong());
+    }
+
+    @Test
+    void testFindAllReturnsEmptyListOfOrders() {
+        // Given
+        List<Order> orders = List.of();
+
+        // When
+        when(orderRepository.findAll())
+                .thenReturn(orders);
+
+        List<Order> response = orderService.findAll();
+
+        // Then
+        assertSame(orders, response);
+        assertTrue(response.isEmpty());
+        verify(orderRepository).findAll();
+    }
+
+    @Test
+    void testFindAllReturnsListOfOrders() {
+        // Given
+        List<Order> orders = List.of(
+                new Order("14 Green Street"),
+                new Order("21 1/2 Argument Street"),
+                new Order("Flat 2, 24B Oswald Street")
+        );
+
+        // When
+        when(orderRepository.findAll())
+                .thenReturn(orders);
+
+        List<Order> actualFindAllResult = orderService.findAll();
+
+        // Then
+        assertSame(orders, actualFindAllResult);
+        assertFalse(actualFindAllResult.isEmpty());
+        verify(orderRepository).findAll();
+    }
+
+    @Test
+    void testAddItemToBasketIsSuccessful() {
+        // Given (Setup)
+        Order order = new Order();
+        Item item = new Item("King Burger", "A burger fit for a king.", new BigDecimal("12.99"), true);
+        Order expectedResponse = new Order("42 Main St");
+        expectedResponse.setOrderItems(Set.of(new OrderItem(order, item, 1)));
+
+        // When (Mocking, sending requests)
+        when(orderRepository.findById(anyLong()))
+                .thenReturn(Optional.of(order));
+        when(itemService.findById(anyLong()))
+                .thenReturn(item);
+        when(orderRepository.save(any()))
+                .thenReturn(expectedResponse);
+
+        Order response =
+                orderService.addItemToBasket(12L, 12L, 1);
+
+        // Then (assertions)
+        assertEquals(expectedResponse, response);
```

Comment:
Since expectedResponse is not null and we are asserting that it is equals to response then the next line wouldn't make sense as if it passes line 138 the next line will be always false hence we can remove it or alternatively put it before line 138.

Reply:
Should be resolved now!